#!/usr/bin/env python
PACKAGE = "dynamic_tutorials"
#初始化ROS，并导入参数生成器
from dynamic_reconfigure.parameter_generator_catkin import *
#初始化参数生成器
gen = ParameterGenerator()
#定义需动态配置的参数，格式为
#gen.add(name , type, level, description, default, min, max)

gen.add("Omega_p1",    double_t,    0, "Omega_p1", 1,  0, 100)
gen.add("Omega_p2", double_t, 0, "Omega_p2",   1,  0, 100)
gen.add("Omega_p3",    double_t,    0, "Omega_p3",  1,  0, 100)
gen.add("Lambda_p.diagonal1",   double_t,   0, "Lambda_p.diagonal1", 1,  0, 100)
gen.add("Lambda_p.diagonal2", double_t, 0, "Lambda_p.diagonal2",    1,  0, 100)
gen.add("Lambda_p.diagonal3",    double_t,    0, "Lambda_p.diagonal3",  1,  0, 100)
gen.add("K_v.diagonal1",   double_t,   0, "K_v.diagonal1",  1,  0, 100)
gen.add("K_v.diagonal2", double_t, 0, "K_v.diagonal2",    1,  0, 100)
gen.add("K_v.diagonal3",    double_t,    0, "K_v.diagonal3",  1,  0, 100)


#生成必要文件并退出
exit(gen.generate(PACKAGE, "dynamic_tutorials", "Tutorials"))


# #!/usr/bin/env python
# PACKAGE = "dynamic_tutorials"
# #初始化ROS，并导入参数生成器
# from dynamic_reconfigure.parameter_generator_catkin import *
# #初始化参数生成器
# gen = ParameterGenerator()
# #定义需动态配置的参数，格式为
# #gen.add(name , type, level, description, default, min, max)
# gen.add("int_param",    int_t,    0, "An Integer parameter", 50,  0, 100)
# gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
# gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)
# #生成一个枚举类型的值
# size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),                      
# 					   gen.const("Medium",     int_t, 1, "A medium constant"),                      
# 					   gen.const("Large",      int_t, 2, "A large constant"),                      
# 					   gen.const("ExtraLarge", int_t, 3, "An extra large constant")], 
# 					   "An enum to set size")
# gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)
# #生成必要文件并退出
# exit(gen.generate(PACKAGE, "dynamic_tutorials", "Tutorials"))


